File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/infrastructure/repository/PostJpaRepository.java
package com.eager.questioncloud.core.domain.post.infrastructure.repository;

import com.eager.questioncloud.core.domain.post.infrastructure.entity.PostEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface PostJpaRepository extends JpaRepository<PostEntity, Long> {
    Optional<PostEntity> findByIdAndWriterId(Long id, Long userId);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/infrastructure/repository/PostJpaRepository.kt
package com.eager.questioncloud.core.domain.post.infrastructure.repository

import com.eager.questioncloud.core.domain.post.infrastructure.entity.PostEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface PostJpaRepository : JpaRepository<PostEntity, Long> {
    fun findByIdAndWriterId(id: Long, userId: Long): Optional<PostEntity>
}
